mysql-test/suite/binlog_encryption/t/binlog_rotate_key.test

```diff
@@ -0,0 +1,154 @@
+--source include/have_debug.inc
+
+--echo # Currently binlog is disabled. After initializing check_consecutive_binlog_key_versions we will restart the
+--echo # server with binlog enabled. check_consecutive_binlog_key_versions will make sure that a new version of
+--echo # percona_binlog key is used after each rotate_system_key followed by FLUSH LOGS calls.
+SET SESSION debug="+d,check_consecutive_binlog_key_versions";
+
+--let $restart_parameters=restart:--log-bin=master-bin
+--source include/restart_mysqld.inc
+
+--echo # master-bin.000002
+
+SELECT 1;
+SELECT 2;
+
+--let $assert_text= Successfull rotation of percona_binlog
+--let $assert_cond= "[SELECT rotate_system_key\\(\\"percona_binlog\\"\\) = 1]" = 1
+--source include/assert.inc
+
+FLUSH LOGS;
+
+--echo # master-bin.000003
+
+SELECT 3;
+SELECT 4;
+
+--let $assert_text= Successfull rotation of percona_binlog
+--let $assert_cond= "[SELECT rotate_system_key\\(\\"percona_binlog\\"\\) = 1]" = 1
+--source include/assert.inc
+
+FLUSH LOGS;
+
+--echo # master-bin.000004
+
+FLUSH LOGS;
+SELECT 5;
+SELECT 6;
+
+FLUSH LOGS;
+
+--echo # master-bin.000005
+
+SET SESSION debug="-d,check_consecutive_binlog_key_versions";
+
+# First check that each binlog is encrypted by trying to decrypt it using mysqlbinlog
+let datadir=`select @@datadir`;
+let filename= master-bin.000002;
+let local=$datadir/$filename;
+let outfile=$MYSQLTEST_VARDIR/tmp/binlog_master;
```

Comment:
If this is a sink for output with no care for actual contents, then /dev/null ?

Reply:
Done